api/include/opentelemetry/context/runtime_context.h

```diff
@@ -108,15 +146,46 @@ class RuntimeContext
     return temp_context.GetValue(key);
   }
 
-protected:
-  // Provides a token with the passed in context
-  Token CreateToken(Context context) noexcept { return Token(context); }
+  /**
+   * Provide a custom runtime context storage.
+   *
+   * This provides a possibility to override the default thread-local runtime
+   * context storage. This has to be set before any spans are created by the
+   * application, otherwise context state information will be discarded, which
+   * might lead to unexpected results.
+   *
+   * @param storage a custom runtime context storage
+   */
+  static void SetRuntimeContextStorage(nostd::shared_ptr<RuntimeContextStorage> storage) noexcept
```

Comment:
Is this method typically gonna be the very first method called before any telemetry is emitted by the process?

Reply:
Hard to say. I think it is, but I'm not sure about all the different scenarios.

Reply:
> Is this method typically gonna be the very first method called before any telemetry is emitted by the process?

I documented that this method has to be called before any spans are created.